model_compression_toolkit/core/common/quantization/quantization_params_generation/power_of_two_selection.py

```diff
@@ -0,0 +1,167 @@
+        max_value: Max value (not used for this method).
+        constrained: Whether the threshold should be constrained or not (not used for this method).
+        n_iter: Number of iteration ot search for the threshold (not used for this method).
+        min_threshold: Minimal threshold to use if threshold is too small (used only for kl threshold selection).
+        quant_error_method: an error function to optimize the parameters' selection accordingly.
+
+    Returns:
+        Optimal threshold to quantize the histogram a power of 2 manner.
+    """
+    if quant_error_method == qc.QuantizationErrorMethod.NOCLIPPING:
+        tensor_max = np.max(np.abs(bins))
+        threshold = power_of_two_constraint(tensor_max, min_threshold)
+    else:
+        error_function = get_threshold_selection_histogram_error_function(quant_error_method, p)
+        threshold = qparams_selection_histogram_search(error_function,
+                                                       bins,
+                                                       counts,
+                                                       n_bits,
+                                                       constrained=constrained,
+                                                       n_iter=n_iter,
+                                                       min_threshold=min_threshold)
+    return {THRESHOLD: threshold}
+
+
+def get_threshold_selection_tensor_error_function(quant_error_method: qc.QuantizationErrorMethod,
+                                                  p: int,
+                                                  norm: bool = False,
+                                                  n_bits: int = 8,
+                                                  signed: bool = True) -> Callable:
+    """
+    Returns the error function compatible to the provided threshold method,
+    to be used in the threshold optimization search for tensor quantization.
+    Args:
+        quant_error_method: the requested error function type.
+        p: p-norm to use for the Lp-norm distance.
+        norm: whether to normalize the error function result.
+        n_bits: Number of bits to quantize the tensor.
+        signed: signed input
+
+    Returns: a Callable method that calculates the error between a tensor and a quantized tensor.
+    """
+    quant_method_error_function_mapping = {
+        qc.QuantizationErrorMethod.MSE: lambda x, y, threshold: compute_mse(x, y),
+        qc.QuantizationErrorMethod.MAE: lambda x, y, threshold: compute_mae(x, y),
+        qc.QuantizationErrorMethod.LP: lambda x, y, threshold: compute_lp_norm(x, y, p=p, norm=norm),
+        qc.QuantizationErrorMethod.KL: lambda x, y, threshold: _kl_error_function(x, range_min=0 if not signed else -threshold, range_max=threshold, n_bits=n_bits)
+    }
+
+    return quant_method_error_function_mapping[quant_error_method]
+
+
+def get_threshold_selection_histogram_error_function(quant_error_method: qc.QuantizationErrorMethod,
```

Comment:
Same here, regarding the code duplication

Reply:
done